### PR TITLE
Add line break after first open paren in multiline function call

### PR DESCRIPTION
**Pull request by rotu**
as per developer guide:
https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#open-versus-cuddled-braces
see https://github.com/ament/ament_lint/pull/148

Signed-off-by: Dan Rose <dan@digilabs.io>

### CODE REVIEW

File: rclcpp/test/test_client.cpp [hunk 1]
```diff
@@ -75,7 +75,8 @@ TEST_F(TestClient, construction_and_destruction) {
   }
 
   {
-    ASSERT_THROW({
+    ASSERT_THROW(
+    {
```
Comment (dirk-thomas) [inline rclcpp/test/test_client.cpp]: It looks weird that this is not indented. I guess you ran the linter on this patch and only this version passed?

File: rclcpp/src/rclcpp/timer.cpp [hunk 2]
```diff
@@ -110,8 +110,9 @@ std::chrono::nanoseconds
 TimerBase::time_until_trigger()
 {
   int64_t time_until_next_call = 0;
-  if (rcl_timer_get_time_until_next_call(timer_handle_.get(),
-    &time_until_next_call) != RCL_RET_OK)
+  if (rcl_timer_get_time_until_next_call(
```
[the remark this answers is not in the document]
Comment (rotu) [reply rclcpp/src/rclcpp/timer.cpp]: right you are! opened https://github.com/uncrustify/uncrustify/issues/2360 and I'll get out my trusty regex skills to find it in the codebase :-)

[the remark this answers is not in the document]
Comment (rotu) [reply rclcpp/src/rclcpp/timer.cpp]: done and put in an upstream PR to make this possible in our own rules:
https://github.com/uncrustify/uncrustify/pull/2361

File: rclcpp/src/rclcpp/parameter_client.cpp [hunk 3]
```diff
@@ -71,10 +71,11 @@ AsyncParametersClient::AsyncParametersClient(
   node_services_interface->add_client(set_parameters_base, nullptr);
 
   set_parameters_atomically_client_ =
-    Client<rcl_interfaces::srv::SetParametersAtomically>::make_shared(node_base_interface.get(),
-      node_graph_interface,
-      remote_node_name_ + "/" + parameter_service_names::set_parameters_atomically,
-      options);
+    Client<rcl_interfaces::srv::SetParametersAtomically>::make_shared(
+    node_base_interface.get(),
```
[the remark this answers is not in the document]
Comment (rotu) [reply rclcpp/src/rclcpp/parameter_client.cpp]: No, and I have no idea why. Uncrustify is a fickle mistress

**Comment by rotu** [comment]
> Did you run the tests locally and they passed?

Yep.

**Comment by dirk-thomas** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7796/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3874/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6363/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7630/)

**Comment by rotu** [comment]
Needed a rebase

**Comment by dirk-thomas** [comment]
See updated builds in the comment above.

File: rclcpp/test/test_client.cpp (hunk 1 above)
Comment (rotu) [reply rclcpp/test/test_client.cpp]: As it turns out, this is how I have to leave it. I tried intend_paren_open_brace, but this changed too much other stuff. As it turns out, this will be fixed by the other ament_uncrustify PR, but will involve some breaking PRs.

**Comment by rotu** [comment]
I had to revert the changes to indent_paren_open_brace. How do I rerun CI?

**Comment by dirk-thomas** [comment]
First you will have to resolve the conflicts in this branch and make sure it is rebased to the latest state. I can retrigger the CI build then.

**Comment by rotu** [comment]
resolved, rebased, and squashed

**Comment by dirk-thomas** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7805/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3883/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6372/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7639/) (known test failures on master)

File: rclcpp/test/test_node.cpp [hunk 4]
```diff
@@ -1445,9 +1474,10 @@ TEST_F(TestNode, set_parameters_atomically_undeclared_parameters_not_allowed) {
       };
     node->set_on_parameters_set_callback(on_set_parameters);
 
-    auto ret = node->set_parameters_atomically({
+    auto ret = node->set_parameters_atomically(
+    {
       {name1, 2},
-      {name2, false},  // should fail to be set, failing the whole operation
+      {name2, false},   // should fail to be set, failing the whole operation
```
Comment (dirk-thomas) [inline rclcpp/test/test_node.cpp]: Why is this necessary?

Same below.

Comment (rotu) [reply rclcpp/test/test_node.cpp]: I'm not sure how that snuck in there, but of course I blame uncrustify.

File: rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp [hunk 5]
```diff
@@ -831,25 +829,27 @@ NodeParameters::list_parameters(const std::vector<std::string> & prefixes, uint6
...
+        return false;
+      });
     if (get_all || prefix_matches) {
       result.names.push_back(kv.first);
       size_t last_separator = kv.first.find_last_of(separator);
       if (std::string::npos != last_separator) {
         std::string prefix = kv.first.substr(0, last_separator);
-        if (std::find(result.prefixes.cbegin(), result.prefixes.cend(), prefix) ==
+        if (
+          std::find(result.prefixes.cbegin(), result.prefixes.cend(), prefix) ==
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp]: Why was this changed in https://github.com/ros2/rclcpp/commit/7c031a0ad1561095b0c2c3228a6101880378094d#diff-43c1c52da4c73f0a25534075d66ece63 ?

Comment (rotu) [reply rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp]: Because it's an unnecessary line split that I realized when verifying whether the additional pre-comment whitespace was needed.

**Comment by dirk-thomas** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7817/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3891/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6379/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7649/)

**Comment by dirk-thomas** [comment]
Thanks for iterating on this.

**Comment by rotu** [comment]
Thank *you* for bearing with me. Between unfamiliar tools, multiple codebases, and my naivete to collaborative development, I'm really glad you've been so patient and helpful!
